src/modules/pass.ts

```diff
@@ -0,0 +1,588 @@
+    if (export_children) {
+      children = this.children.map(function (c) {
+        return c.toObject(false);
+      });
+    }
+    return {
+      id: this.id,
+      parent: (this.parent === null) ? null : this.parent.toObject(false),
+      isLeaf: this.isLeaf(),
+      isField: this.isField(),
+      hasFields: this.hasFields(),
+      fullKey: this.fullKey(),
+      children: children
+    };
+  }
+}
+
+
+export class Pass {
+
+  static _items: Item[] = [];
+  static _rootItems: Item[] = [];
+
+  static env = class {
+    static _environment: Dict<string> = {};
+
+    static exists(key: string) {
+      return this._environment.hasOwnProperty(key);
+    }
+
+    static get(key: string) {
+      if (this.exists(key)) return this._environment[key];
+      else return "";
+    }
+  };
+
+  static init_env() {
+    return browser.runtime.sendNativeMessage("passff", {command: "env"})
+      .then((result: Dict<string>) => {
+        Pass.env._environment = result;
+      });
+  }
+
+  static init() {
+    return this.init_env().then(() => {
+      return Pass.initItems();
+    });
+  }
+
+  static initItems() {
+    return this.executePass([]).then(((result: HostAppReturnValue) => {
+      if (result.exitCode !== 0) {
+        return;
+      }
+
+      this._rootItems = [];
+      this._items = [];
+
+      let stdout = result.stdout;
+      // replace utf8 box characters with traditional ascii tree
+      stdout = stdout.replace(/[\u2514\u251C]\u2500\u2500/g, '|--');
+      //remove colors
+      stdout = stdout.replace(/\x1B\[[^m]*m/g, '');
+
+      let lines = stdout.split('\n');
+      let re = /(.*[|`;])+-- (.*)/;
+      let curParent = null;
+
+      for (let i = 0; i < lines.length; i++) {
+        let match = re.exec(lines[i]);
+
+        if (!match) {
+          continue;
+        }
+
+        let curDepth = (match[1].replace('&middot;', '`').length - 1) / 4;
+        let key = match[2].replace(/\\ /g, ' ').replace(/ -> .*/g, '');
+
+        while (curParent !== null && curParent.depth >= curDepth) {
+          curParent = curParent.parent;
+        }
+
+        let item: Item = new Item(curDepth, key, curParent, this._items.length);
+
+        if (curParent !== null) {
+          curParent.children.push(item);
+        }
+
+        curParent = item;
+        this._items.push(item);
+
+        if (item.depth === 0) {
+          this._rootItems.push(item);
+        }
+      }
+      log.debug('Found Items', this._rootItems);
+    }).bind(this));
+  }
+
+  static getPasswordData(item: Item): Promise<Dict<string>> {
+    let result: Dict<string> = {};
+
+    if (item.isLeaf()) { // multiline-style item
+      let args = [item.fullKey()];
+      return this.executePass(args).then((executionResult) => new Promise((resolve, reject) => {
+        let gpgDecryptFailed = executionResult.stderr
+            .indexOf('gpg: decryption failed: No secret key') >= 0;
+
+        while (executionResult.exitCode !== 0 && gpgDecryptFailed) {
```

Comment:
Oh, I also stumbled over this. 
I guess that could result in an infinite loop and should be removed, shouldn't it?